### ISSUE TITLE
HTML ocamlformatter

### ISSUE THREAD

**Issue by AdamBrodzinski** [yawaramin/dream-html#13]
Hello,

I was wondering if it's possible to get the OCaml formatter to change the formatting so that it's the same as your Readme example? I found that your .ocamlformat file gets it closer than the default.

Ideally for me I would love to format it like the example below, where the list only breaks if it can't fit on the line, and the closing list bracket lines up with the function name.

Do you happen to know if either of these possible with the current ocamlformat? If not I may try to open an issue on the ocamlformat repo if it's not already in flight.

Also thanks so much for this library, it's perfect!

```ocaml
nav [class_ "container"] [
  ul [] [txt "Acme"];
  ul [] [
    li [] [a [href "/admin/foo"] [txt "Foo"]];
    li [] [a [href "/admin/bar"] [txt "Bar"]];
    li [] [a [href "/admin/baz"] [txt "Baz"]];
  ];
]
```

**Comment by yawaramin** [comment]
Hi, thanks for the kind words! Unfortunately the closest I've been able to get to a nice style with ocamlformat is the flags that I currently have set. I've played around with pretty much all the flags, but they all change the formatting in undesirable ways. E.g., this looks promising at first:

```
--dock-collection-brackets
  Dock the brackets of lists, arrays and records, so that when the collection does not fit on a single line the brackets are opened on the preceding line and closed on the following line. The flag is set by default.
```

But then you realize it keeps the _opening_ bracket on a line by itself as well, i.e.

```ocaml
nav
  [class_ "container"]
  [
    ...
  ]
```

Nowadays I've actually given up on the idea of customizing ocamlformat in every single OCaml project and reluctantly accepted that something is better than nothing, and just sticking with the default (an empty `.ocamlformat` file) is good enough.
